packages/core-ethereum/src/indexer/index.spec.ts

```diff
@@ -123,6 +123,11 @@ const createChainMock = (
   } as unknown as ChainWrapper
 }
 
+const assertLater = async (assertion: () => any, delay: number) => {
```

Comment:
What about `await`ing the end of the Node.js event loop and repeat the assertion check on every end of the end loop until we reach a certain timeout?

The unit tests could then run without "randomly chosen" timeouts and be quicker since they don't need to `await` the entire timeout.

This one could help: https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/#setimmediate-vs-settimeout

Reply:
we randomly choose the timeout too, let's merge this and u can make a follow up PR with ur suggestion

Reply:
Sounds good, this approach might any need some exploration

Reply:
not loving the approach of waiting an arbitrart timeout and trying things - this is how we get flaky tests - is there not a way we can wait until a specific event (block is mined etc) so we can _know_ something is done.

Reply:
all tests have a default timeout of 2 seconds, we _need_ a timeout, I agree that we can improve how we _know_ something is done through adding an event, but the timeout must remain

Reply:
- added events `status` and `block-processed`
- used defer to await for these events